.circleci/config.yml

```
@@ -262,35 +259,27 @@ jobs:
             . ./bin/env
             $GCLOUD compute instances delete ${INSTANCE_NAME} --zone ${ZONE}
 
-  generate-artifacts:
+  check-new-artifacts:
     docker:
       - image: ghcr.io/cybozu/golang:1.22-jammy
     steps:
       - checkout
-      - run: go install ./pkg/generate-artifacts/
-      - run: generate-artifacts --release > artifacts_release.go
-      - persist_to_workspace:
-          root: .
-          paths:
-            - "artifacts_release.go"
+      - run: make update-artifacts
+      - run: 
+          name: Compare artifacts
+          command: |
+            diffs=$(git diff --name-only main)
+            if [ "$diffs" = "" ]; then printf "%s\n" "$diffs"; exit 0; fi
```

The comparison seems inverted.

```suggestion
            diffs=$(git diff --name-only main)
            if [ "$diffs" != "" ]; then printf "%s\n" "$diffs"; exit 0; fi
```

And it could be simplified.

```suggestion
            git --no-pager diff --name-only main
```

I suppose that this `run` step is just for information.
We don't want to stop the sequence even if there were differences, right?

`git diff` without `--exit-code` option returns zero even if there were differences.
You can directly use this command.
In case you didn't know, if `git diff` returned non-zero, the assignment `diffs=$(git diff ...)` also returns non-zero and fails.

`--no-pager` may not be necessary.